application.go

```diff
@@ -201,12 +211,9 @@ func (a *application) Run() {
 		panic(fmt.Errorf(setupRootCommandNotCalledError))
 	}
 
-	cmd := a.root
-	log := a.state.Logger
-
```

Comment:
I found this while troubleshooting the exit event work -- it is not guaranteed that at assignment there will be a logger, thus, when a logger is assigned later we will not have that reference and will attempt to call a method on nil. This change uses the objects directly off of the application instance to prevent this.